node/sc/main_bridge_handler.go

```diff
@@ -109,18 +110,24 @@ func (mbh *MainBridgeHandler) handleServiceChainTxDataMsg(p BridgePeer, msg p2p.
 	}
 
 	// Only valid txs should be pushed into the pool.
-	validTxs := make([]*types.Transaction, 0, len(txs))
-	// validTxs := []*types.Transaction{}
-	var err error
-	for i, tx := range txs {
+	// Invalid txs found are sent to child chain for further action.
+	invalidTxs := make([]InvalidParentChainTx, 0)
```

Comment:
Why are. you remove the validTxs?
You are doing `AddRemote` and `SendServiceChainInvalidTxResponse` in every iteration rather doing `AddRemotes` for lazy processing.
 Do you prioritize `addRemote` than sending error response so that the validTxs are processed with a little faster?

Reply:
I got it. Thanks for your explanation.